fakebook/src/app/services/auth.service.ts

```diff
@@ -14,17 +14,31 @@ export class AuthService {
   isAuthenticated: boolean | unknown;
 
   constructor(public oktaAuth: OktaAuthService, public router: Router) {
+    this.updateAuthState(false);
+    this.oktaAuth.$authenticationState.subscribe((isAuthenticated) =>
+    this.updateAuthState(isAuthenticated)
+    );
   }
 
   updateAuthState(isAuthenticated: boolean): void {
+    this.isAuthenticated = isAuthenticated;
+    if (isAuthenticated) {
+      this.oktaAuth.getUser().then(console.log);
+    }
   }
 
   subscribeAuthStateChange(updateFn: (authState: boolean) => void): void {
```

Comment:
this is not how we interact with observables, you just expose the observable object itself for others to subscribe on, we don't need to put a callback layer around the observables.